addons/website/tests/test_ui.py

```diff
@@ -317,3 +317,36 @@ def test_29_website_backend_menus_redirect(self):
         self.env.ref('base.user_admin').action_id = self.env.ref('base.menu_administration').id
         self.assertFalse(menu_root.action, 'The top menu should not have an action (or the test/tour will not test anything).')
         self.start_tour('/', 'website_backend_menus_redirect', login='admin')
+
+    def test_website_no_dirty_page(self):
+        # Having enough menus to trigger the "+" folded menus is known to
+        # wrongfully mark the page as dirty. So what should be done here is
+        # creating a lot of menus first. Actually there are 3 cases:
+        #
+        # - the menu is not folded outside of edit mode and when entering edit
+        #   mode, the "+" appears and some menu are folded
+        #
+        # - the menu is folded outside of edit mode and when entering edit mode
+        #   the resize actually makes it so a different set of menu is folded
+        #
+        # - the menu is folded outside of edit mode and when entering edit mode
+        #   it stays the same (known to have been broken because edit mode
+        #   tweaks the dropdown behavior)
+        #
+        # So something like this could be done:
+        # website = self.env['website'].get_current_website()
+        # self.env['website.menu'].create([{
+        #     'name': 'Menu %s' % i,
+        #     'website_id': website.id,
+        #     'parent_id': website.menu_id.id,
+        # } for i in range(20)])
+        #
+        # Unfortunately, in stable, the dirty check could not be reviewed
+        # enough safely to make this case work. For now, we just remove most
+        # menu items during this test to make sure it passes. It will be fixed
+        # in further version and potentially backported (TODO).
+        website = self.env['website'].get_current_website()
+        menus = self.env['website.menu'].search([('parent_id', '=', website.menu_id.id)])
+        menus[1:].unlink()
```

Comment:
```suggestion
        menus = website.menu_id.child_id[1:].unlink()
```
(Yeah, `child_id` is one of the many wrongly named fields in Odoo)

Reply:
Don't know how I did not see it when confirming parent_id existed, it is right below.

Reply:
The bad name I guess, missing a `s` at the end, confusing as hell